api/routes.py

```diff
@@ -135,15 +135,16 @@ def build_request(task_data):
 
 def get_current_user():
   """Get the username of currently logged-in user."""
-  from app import oauth
   try:
+    # Importing the oauth early results in an error
```

Comment:
Not sure why you are getting an error here, putting the import at the top works for me.

Reply:
Hmm, I keep getting `ImportError: cannot import name 'oauth' from partially initialized module 'app' (most likely due to a circular import) (/app/app.py)`

Reply:
Oops, I just realized I did my testing on a branch where I had slightly refactored the extensions: addressing https://phabricator.wikimedia.org/T330263 should fix this, and then you could import oauth with`from api import oauth`. Sorry for the confusion!